modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -142,7 +150,51 @@
 		},
 
 		/**
-		 * Checks if the pasted data is image and passes it to
+		 * Handler for when images are uploaded
+		 *
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _onImageUploaded
+		 * @param {Object} event Event data
+		 * @protected
+		 */
+		_onImageUploaded(event) {
```

Comment:
In the plugin, the [code that handles single image src rewriting](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js#L323-L340) is in `_onUploadComplete`. The code in this PR that does the same type of change, but on complex html, is in `_onImageUploaded`. We should have these two in one place. A single image could be just a simple case of HMTL handling.

Reply:
I'll look into it, but rewriting an image's  `src` attribute is not the same thing as rewriting the `src` attributes for images in a `fragment` which is what we get when we paste.

Reply:
What's the difference? I was under the assumption that fragment elements are just CKEditor's wrapper around HTML elements, similar to AUI node. Does something additional or different happen when you set `src` to fragment element?

Reply:
Another question: how do access the editor from our `_onImageUploaded` function if it doesn't receive the event anymore? 
We don't have any reference to it

Reply:
Answering to myself: I'll guess I'll pass it around to the `_onImageUploaded` function directly but I don't like that. I don't see what wrong with event based programming